### ISSUE TITLE
Add References attached to transactions

### ISSUE THREAD

**Issue by igormcoelho** [neo-project/neo#444]
I think it would be very useful to allow transactions to attach References (Coin References), without a corresponding output. Usually, References are taken from Inputs, which are consumed and correspond to outputs. The idea here is to just attach References that are not Inputs to be consumed, just to be validated.
So, the contribution of these references will be to enrich GetScriptsForVerifying, and to easily allow the implementation of features such as Staked Tx, since all references will be naturally validated (plugin will only guarantee that references have enough assets to backup tx).

**Comment by igormcoelho** [comment]
@erikzhang I would like to include this together with PR https://github.com/neo-project/neo/pull/443, making it available on transaction version 2. 
@shargon Using Block Height and References, we can easily build staked tx and PoW for tomorrow if necessary.

**Comment by shargon** [comment]
I agree, I should be able to do a manual TX without the chain and the References made it more difficult. But in neo 3.0 this should be eliminated.
